src/librustc/ty/error.rs

```diff
@@ -58,6 +58,8 @@ pub enum TypeError<'tcx> {
     ConstMismatch(ExpectedFound<&'tcx ty::Const<'tcx>>),
 
     IntrinsicCast,
+    /// Safe `#[target_feature]` functions are not assignable to safe function pointers.
+    TargetFeatureCast(Vec<Span>),
```

Comment:
Hmm, maybe store a `Box<[Span]>` instead as we're never mutating the collection.

Reply:
The `TargetFeatureCast` is created here:

https://github.com/rust-lang/rust/blob/553e7ab7d997f72588c107c2346934c373c55005/src/librustc_typeck/check/coercion.rs#L706-L713

So I'm not sure if storing a slice is a good idea, nor do I know precisely how to do it.

Reply:
Given `impl<A> FromIterator<A> for Box<[A]>` you don't need to change that code; just change the stored type and it should just work.

Reply:
I didn't know `Box` implemented `FromIterator` :eyes:. Makes a lot more sense now 😅